### DISCUSSION TITLE
How can I isolate "proper" photo albums?

### DISCUSSION THREAD

**Discussion by jfmoyen** [deckerst/aves#823] [Q&A]
Good morning,

I recently discovered Aves, that looks like the Gallery app I need - and it's open source. So far so good, and many thanks for the great work!

There is however something that puzzles me and does not quite work for me. The "tier" system is a great idea, but it brings unexpected results to me.

At the moment I have three tiers active : "Common" (with Download and Screenshot); "Apps" (with things such as WhatsApp, to which I manuallly added the other picture apps I routinely use -- snapseed, my camera download app, and so on ); and "others".

"Others" is my source of gripe. It is a pell-mell of everything, including (1) albums created by Aves (in .../Pictures/myAlbum); (2) albums from the original (Miui in my case) gallery app; (3) every other folder containing image found on my system, for instance the cover image of every music album on my device, the assets of half of the apps (so I have a folder called "Button/img" and "ColorPicker/img" for instance, from DroidScript as it where), etc. I have perhaps ~40 folders with "proper" photos, but something like 80 or 100 in "Others", which is inconvenient to say the least as I have to sieve through a host of junk everythime I look for an album (well, not junk exactly, but not a "photo" in the sense of an image I want to look at....).

Now, it seems straightforwards to me that there should be some separation between "legitimate" photo folders (at least those created by aves itself and found in its own directory) and everything else - it may be useful to have access to them, yes, but if I want to see my photos from, say, "Canada trip 1756" I do not want to look for them between "ColorPicker" and "Button" and scroll past "AppData" "Abbey Road 1969" and "Brandenburger Concerto" before getting there... 

Obviously, there are cures such as manually hiding each and every unwanted folder (not ideal, I would prefer to have them visibile but in their own rubric, and that would not be portable when changing device). In fact, it seems to me that "others" should only apply to the media files that are not "legitimate" photos (assets from apps, etc), and there should be a tier for "real" photo albums, if that makes sense.

I have a similar issue with setting up google image saves, so it may have to do with the way my phone is set-up ... I'd value any help, nonetheless.

So I don't know if this is a feature request, perhaps to be merged with those concerning "tiers" and "nested folders", or rather a plea for help with broader device settings...

Thanks !

**Comment by deckerst** [comment]
The default tiers are a way to highlight common folders that are found on nearly all devices, like Downloads, Camera, etc. and it's based on a few assumptions. Beyond that I cannot assume much about the location of "proper" photos, because Android devices are widely diverse.

Music folders containing album cover images are typically all located within the same root folder, so you can hide in Aves that root folder, without having to hide every single individual folder containing an album cover. To do so, go to `Settings` > `Privacy` > `Hidden items` > `Hidden paths`, and select the folder at the root of your music collection.

Regarding app assets, I'd blame the app for exposing these assets as public media files, while they likely should not and be kept private to that app. The developer of the app could add a ".nomedia" file next to it. If these apps are open-source, maybe you could suggest this change to them. But as a user, if you want to hide them within Aves, you could do the same thing as for the music collection, if the app assets are all within a single or few root folders.

**Comment by jfmoyen** [comment]
Thanks for the suggestions - they help.
The "faulty" app is Droidscript, which is not open source (although reasonably clean), I'll see if there is a way to reach out to the dev.

I still wonder whether it would not make sense for Aves to give its own folders a special status (= tier). I suppose it depends on your philosophy. Do you want it to be an agnostic picture manager, dealing with everything it finds on the device ? Or rather a one-stop solution, also serving with a default storage place for photos ?

My workflow would sit better with the second option - I have photos from different sources (phone, a drone, an external camera) which I sort and process (minimally, with snapseed) before I organize them in their "final" folders. Thus, I have a rather restricted set of folders that should be first-class citizens (the ones where my "final" photos are stored, as opposed to everything else that is essentially of temporary nature).

**Reply by deckerst** [reply]
> Droidscript

Out of curiosity, do you use this IDE on a phone or a tablet? Wouldn't it be difficult to write code on a phone?

> Do you want it to be an agnostic picture manager

Yes, especially because there is nothing special to what you refer to as Aves "own folders". Albums created within Aves are simply placed in the most appropriate place on storage, which is the "Pictures" root folder. Nothing special beyond this.

> I have a rather restricted set of folders that should be first-class citizens

Maybe pinning these folders would be a suitable solution for you? You can pin albums by selecting them (long press on the thumbnail or via the top-right menu), then tap on the `pin` icon at the top.

**Comment by jfmoyen** [comment]
> > Droidscript
> 
> Out of curiosity, do you use this IDE on a phone or a tablet? Wouldn't it be difficult to write code on a phone?

Yes it is a pain (although a bluetooth keyboard would help). I actually code from computer, using the (rather clever) WiFi IDE. Basically your device works as a web server to which you connect from your browser.

> > I have a rather restricted set of folders that should be first-class citizens
> 
> Maybe pinning these folders would be a suitable solution for you? You can pin albums by selecting them (long press on the thumbnail or via the top-right menu), then tap on the pin icon at the top.

Ah, that works, good idea ! 

Thanks
